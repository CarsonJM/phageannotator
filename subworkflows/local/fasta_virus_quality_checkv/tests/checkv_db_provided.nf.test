nextflow_workflow {

    name "Test Subworkflow: FASTA_VIRUS_QUALITY_CHECKV"
    script "subworkflows/local/fasta_virus_quality_checkv/main.nf"
    workflow "FASTA_VIRUS_QUALITY_CHECKV"
    tag "subworkflows"
    tag "subworkflows_local"
    tag "fasta_virus_quality_checkv"
    tag "fasta_virus_quality_checkv_db_provided"


    test("Parameters: --checkv_db provided") {

        when {
            params {
                load("$baseDir/tests/nf-test-local-db-params.json")
                outdir              = "$outputDir"
            }
            workflow {
                """
                input[0] = Channel.of([ [ id:'test' ], [ file('https://github.com/nf-core/test-datasets/raw/modules/data/genomics/prokaryotes/bacteroides_fragilis/illumina/fasta/test1.contigs.fa.gz', checkIfExists: true) ] ])
                input[1] = Channel.of([ file(params.checkv_db, checkIfExists:true ) ])
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(workflow.out).match() }
            )
        }
        cleanup {
            path("$workDir").deleteDir()
        }
    }
}
