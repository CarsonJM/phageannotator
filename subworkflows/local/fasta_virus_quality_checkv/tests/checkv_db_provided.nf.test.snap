{
    "Parameters: --checkv_db provided": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "viruses.fna.gz:md5,67028c9f9447855f86013fd43a5f6105"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test"
                        },
                        "proviruses.fna.gz:md5,f2b8287ef49757e3a0320217b8add37d"
                    ]
                ],
                "2": [
                    [
                        {
                            "id": "test"
                        },
                        "test_quality_summary.tsv:md5,04e827b7efefe52727a904cfd3af1639"
                    ]
                ],
                "3": [
                    "versions.yml:md5,bde8142f045061369c16cd56d51698ed",
                    "versions.yml:md5,e4d4311cae4bcb6711e82eef25b187ff"
                ],
                "proviruses_fna_gz": [
                    [
                        {
                            "id": "test"
                        },
                        "proviruses.fna.gz:md5,f2b8287ef49757e3a0320217b8add37d"
                    ]
                ],
                "quality_summary_tsv": [
                    [
                        {
                            "id": "test"
                        },
                        "test_quality_summary.tsv:md5,04e827b7efefe52727a904cfd3af1639"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,bde8142f045061369c16cd56d51698ed",
                    "versions.yml:md5,e4d4311cae4bcb6711e82eef25b187ff"
                ],
                "viruses_fna_gz": [
                    [
                        {
                            "id": "test"
                        },
                        "viruses.fna.gz:md5,67028c9f9447855f86013fd43a5f6105"
                    ]
                ]
            }
        ],
        "timestamp": "2023-12-06T15:54:04.516229718"
    }
}